cypress.config.ts

```diff
@@ -83,18 +84,16 @@ export default defineConfig({
           return null;
         },
       });
-      if (config.video) {
-        // delete video if test failed
-        on('after:spec', (spec, results) => {
-          if (results.video) {
-            if (results.stats.failures || results.stats.skipped) {
-              console.log('keeping the video %s', results.video);
-            } else {
-              unlinkSync(results.video);
-            }
+
+      on(
+        'after:spec',
+        (spec: Cypress.Spec, results: CypressCommandLine.RunResult) => {
+          if (results && results.video && results.stats.failures === 0) {
+            if (existsSync(results.video)) unlinkSync(results.video);
```

Comment:
This could use some code comments for those not familiar with cypress configuration.

Reply:
It was actually failing on this after:spec. I couldn't find a way to set the video:true to conditional on the browser as it's not set till after the config is initiated. 

So this was always causing issues when it tried to find the video file, but it didn't exist because Firefox fails at the record step. 

The existsSync step gets around that.